editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/maven/CamelProjectConfigurator.java

```diff
@@ -280,6 +280,16 @@ private String getCamelVersion(MavenProject mavenProject) throws CoreException {
                 }
             }
         }
+        // use deprecated dependency method as a last resort 
+        if (mavenProject.getCompileDependencies() != null) {
+        	Iterator<Dependency> depIter = mavenProject.getCompileDependencies().iterator();
```

Comment:
to stay consistent with the code in the same method, what do you think to avoid usage of an explicit Iterator and use for loop on the list from mavenproject.getCompileDependencies() ?
(it is reducing the number of  lines also)